sql/hive-thriftserver/src/main/java/org/apache/hive/service/auth/HiveAuthFactory.java

```diff
@@ -92,7 +95,26 @@ public String getAuthName() {
   public static final String HS2_PROXY_USER = "hive.server2.proxy.user";
   public static final String HS2_CLIENT_TOKEN = "hiveserver2ClientToken";
 
-  public HiveAuthFactory(HiveConf conf) throws TTransportException {
+  private static Field keytabFile = null;
+  private static Method getKeytab = null;
+  static {
+    Class<?> clz = UserGroupInformation.class;
+    try {
+      keytabFile = clz.getDeclaredField("keytabFile");
+      keytabFile.setAccessible(true);
+    } catch (NoSuchFieldException nfe) {
+      keytabFile = null;
```

Comment:
nit: We should include a debug message (not INFO) indicating the exception.
Same for getKeytab below.

This will help debug in case the signatures of these private methods changes later, and we are not sure what is happening.

Reply:
In the current STS code, we avoid using any log method. You can check the code, all the log method which used before (in Spark1) is removed now.

Reply:
Ok, I just remember Steve commented on some JIRAs about this thing, but I cannot remember where it is. Anyway I will log some logs here.